src/js/main.js

```diff
@@ -11,3 +11,88 @@ var ReactDOM = require('react-dom');
 ///      Edit below. All JS you need is included above.     ///
 ///////////////////////////////////////////////////////////////
 
+var list = [{ key: 0, task:'Finish Time Machine', category: 'regular' }, { key: 1, task: 'Realign Spectral Anamoly Cannon', category: 'regular' }, { key: 2, task: 'Explain Dark Matter to Steven Hawking', category: 'urgent' }, { key: 3, task: 'Do the dishes', category: 'long-term' }];
+    
+var TodoList = React.createClass({
+
+  getInitialState: function () {
+  
+    return { changed: false };
+  
+  },
+
+  render: function() {
+    return (
+    <div className="categoryList">
+        <TodoInput list={this.props.list}/>
+        <TodoItems list={this.props.list}/>
+    </div>
+    );
+  }
+});    
+    
+var TodoInput = React.createClass({
+
+  handleClick: function(e) {
+    
+        e.preventDefault();
+        
+        list.push({key: list.length, task: $('#item-INPUT').val(), category: $('#categorySelect').val() });
+        
+        console.log(list);
+        
+        ReactDOM.render(
+            <TodoList list={list}/>,
+            document.getElementById('content')
+        );
+        
```

Comment:
This was an interesting way of accomplishing this. Not that there is anything wrong with it. 

The way that would be a bit simpler and is a little more of the "React.js" way. Would have been to use state to keep track of the items.

If you use the setState() method then updating the array of items, React will automatically re-render only the parts of the DOM that need to change as opposed to re building the entire DOM as you are here. This would be done in the TodoList component.

Also, if you use a callback function passed to TodoInput from TodoList. This function is the method that React uses to pass data up to parent components. You could let TodoList manage your state, and separate duties. Right now, the TodoInput has a dependency to keep track of the array of items and doesn't fallow the single responsibility rule.

Lastly, you also have a hard dependency on jQuery. React.js can get values from its own shadow DOM via the ref prop.